src/MemoryTracing/MemoryTracingUtils.cpp

```diff
@@ -85,4 +90,139 @@ std::optional<SystemMemoryUsage> GetSystemMemoryUsage() noexcept {
   return parsing_result;
 }
 
+int64_t ParseProcessStatus(const std::string& status_content) {
+  if (status_content.empty()) return kMissingInfo;
+
+  std::vector<std::string> lines = absl::StrSplit(status_content, '\n');
+  if (lines.empty()) return kMissingInfo;
+
+  for (const std::string& line : lines) {
+    if (absl::StartsWith(line, "VmRSS")) {
+      std::vector<std::string> splits = absl::StrSplit(line, ' ', absl::SkipWhitespace{});
+      // According to the kernel code
+      // https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/fs/proc/task_mmu.c,
+      // the size unit of the VmRSS field in the /proc/<PID>/status file is fixed to "kB", which
+      // implies 1024 Bytes. We report in "KB" in ProcessSystemUsage.
+      CHECK(splits.size() == 3 && splits[2] == "kB");
+      int64_t rss_kb;
+      if (absl::SimpleAtoi(splits[1], &rss_kb)) return rss_kb;
+      return kMissingInfo;
+    }
+  }
+
+  return kMissingInfo;
+}
+
+std::optional<ProcessMemoryUsage> GetProcessMemoryUsage(uint32_t pid) noexcept {
+  uint64_t current_timestamp_ns = orbit_base::CaptureTimestampNs();
+
+  // Read the /proc/<pid>/status file and extract the rss value in kilobytes.
+  ErrorMessageOr<std::string> reading_result =
+      orbit_base::ReadFileToString(absl::StrFormat("/proc/%d/status", pid));
+  if (!reading_result.has_value()) {
+    ERROR("%s", reading_result.error().message());
+    return std::nullopt;
+  }
+  int64_t process_rss_kb = ParseProcessStatus(reading_result.value());
+
+  ProcessMemoryUsage process_memory_usage;
+  process_memory_usage.set_pid(pid);
+  process_memory_usage.set_timestamp_ns(current_timestamp_ns);
+  process_memory_usage.set_rss_kb(process_rss_kb);
+
+  return process_memory_usage;
+}
+
+std::string ParseProcessCGroup(const std::string& cgroup_content) {
```

Comment:
Similarly here, make the name more specific to what the function is actually doing.

Reply:
Thanks for the suggestion! Rename it as `GetProcessMemoryCGroupName` now.